pegasus/helpers/form_helpers.rb

```diff
@@ -135,7 +135,12 @@ def insert_form(kind, data, options={})
     updated_ip: request.ip,
   }
   row[:user_id] = dashboard_user[:id] if dashboard_user
-  row[:id] = DB[:forms].insert(row)
+
+  if kind == "HocSignup2015" && DB[:forms].where(email: row[:email], kind: kind, name: row[:name]).count > 0
```

Comment:
This sequence will always do 2 db calls for both updates and insert.

I think db.update returns a count of rows updated. What you can do is something like
if (updateRow() != 1) {
insertRow();
}

But the idea situation would be some suppose for upsert / merge.